x-pack/plugins/timelines/public/components/t_grid/body/index.tsx

```diff
@@ -253,45 +286,64 @@ export const BodyComponent = React.memo<StatefulBodyProps>(
       setVisibleColumns(columnHeaders.map(({ id: cid }) => cid));
     }, [columnHeaders]);
 
-    const [leadingTGridControlColumns, trailingTGridControlColumns] = useMemo(
-      () =>
-        [leadingControlColumns, trailingControlColumns].map((controlColumns) =>
-          transformControlColumns({
-            columnHeaders,
-            controlColumns,
-            data,
-            isEventViewer,
-            actionColumnsWidth: hasAdditionalActions(id as TimelineId)
-              ? getActionsColumnWidth(
-                  isEventViewer,
-                  showCheckboxes,
-                  DEFAULT_ICON_BUTTON_WIDTH * NUM_OF_ICON_IN_TIMELINE_ROW + EXTRA_WIDTH
-                )
-              : controlColumns.reduce((acc, c) => acc + (c.width ?? MIN_ACTION_COLUMN_WIDTH), 0),
-            loadingEventIds,
-            onRowSelected,
-            onRuleChange,
-            selectedEventIds,
-            showCheckboxes,
-            tabType,
-            timelineId: id,
-          })
-        ),
-      [
-        columnHeaders,
-        data,
-        id,
-        isEventViewer,
-        leadingControlColumns,
-        loadingEventIds,
-        onRowSelected,
-        onRuleChange,
-        selectedEventIds,
-        showCheckboxes,
-        tabType,
+    const [leadingTGridControlColumns, trailingTGridControlColumns] = useMemo(() => {
+      const CHECK_BOX_COLUMN_WIDTH = 32;
+      const CHECK_BOX_COLUMN_ID = 'checkbox-control-column';
+
+      const checkBoxControlColumn: ControlColumnProps = {
```

Comment:
I am not sure where to place `checkBoxControlColumn`.

Reply:
maybe you can export the whole obkect from `x-pack/plugins/timelines/public/components/t_grid/body/control_columns/index.tsx`